### DISCUSSION TITLE
PEP621 and uv package manager: how to specify extra index URL?

### DISCUSSION THREAD

**Discussion by Gr1N** [renovatebot/renovate#31151] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Hey!

I'm configuring Renovate to work correctly with the internal PyPI server. I have a bunch of Python projects that are under the `uv` package manager. Everything seems to work as expected, except I can't figure out how to configure the `UV_EXTRA_INDEX_URL` env variable for my internal PyPI server.

Because of that, dependency locking falls with error messages like:
```
cmd": "/bin/sh -c uv lock --upgrade",
         "stderr": "Using Python 3.12.5 interpreter at: /opt/containerbase/tools/python/3.12.5/bin/python3\n  × No solution found when resolving dependencies:\n  ╰─▶ Because there is no version of my-package==X.XX.X and your workspace requires my-package==X.XX.X, we can conclude that your workspace's requirements\n      are unsatisfiable.\n",
         "stdout": "",
```

The same I see on my dependencies dashboard:
```
Renovate failed to look up the following dependencies: Failed to look up pypi package my-package
```

I would appreciate your help with this. Thanks!

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Are you asking a question, or reporting a bug?

**Comment by Gr1N** [comment]
It depends; on the first hand, I see that the `uv lock` command [is being called](https://github.com/renovatebot/renovate/blob/25fd14b2f25985824ab8bd503e84b9af25dce44e/lib/modules/manager/pep621/processors/uv.ts#L19) without specifying `--extra-index-url`, so it can be a bug.

On the other one, the same thing can be solved by passing `UV_EXTRA_INDEX_URL` to the environment, but I can't find in the 
Renovate documentation on how to do that. Please let me know if this can be solved without modifying the Renovate codebase.

**Reply by rarkins** [reply]
No, it requires a feature request.
